src/main/java/at/hannibal2/skyhanni/features/combat/BroodmotherFeatures.kt

```diff
@@ -0,0 +1,157 @@
+import at.hannibal2.skyhanni.utils.RenderUtils.renderString
+import at.hannibal2.skyhanni.utils.SimpleTimeMark
+import at.hannibal2.skyhanni.utils.SoundUtils
+import at.hannibal2.skyhanni.utils.SoundUtils.playSound
+import at.hannibal2.skyhanni.utils.StringUtils
+import at.hannibal2.skyhanni.utils.TimeUtils.format
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+import kotlin.time.Duration.Companion.minutes
+import kotlin.time.Duration.Companion.seconds
+import kotlin.time.DurationUnit
+
+@SkyHanniModule
+object BroodmotherFeatures {
+
+    enum class StageEntry(private val str: String, val minutes: Int) {
+        SLAIN("§eSlain", 10),
+        DORMANT("§eDormant", 9),
+        SOON("§6Soon", 6),
+        AWAKENING("§6Awakening", 3),
+        IMMINENT("§4Imminent", 1),
+        ALIVE("§4Alive!", 0);
+
+        override fun toString() = str
+    }
+
+    private val config get() = SkyHanniMod.feature.combat.broodmother
+    private val spawnAlertConfig get() = config.spawnAlertConfig
+
+    private var lastStage: StageEntry? = null
+    private var currentStage: StageEntry? = null
+    private var broodmotherSpawnTime = SimpleTimeMark.farPast()
+
+    @SubscribeEvent
+    fun onTabListUpdate(event: WidgetUpdateEvent) {
+        if (!event.isWidget(TabWidget.BROODMOTHER)) return
+        val newStage = event.widget.matchMatcherFirstLine { group("stage") } ?: ""
+        if (newStage.isNotEmpty()) {
+            lastStage = currentStage
+            currentStage = StageEntry.valueOf(newStage.replace("!", "").uppercase())
+            onStageUpdate()
+        }
+    }
+
+    private fun onStageUpdate() {
+        ChatUtils.debug("New Broodmother stage: $currentStage")
+
+        if (lastStage == null) {
+            onServerJoin()
+            return
+        }
+
+        // ignore if the stage temporarily reverts to Imminent after the Broodmother's death
+        if (currentStage == StageEntry.IMMINENT && lastStage == StageEntry.ALIVE) return
+
+        if (currentStage == StageEntry.ALIVE) {
+            onBroodmotherSpawn()
+            return
+        }
+
+        val timeUntilSpawn = currentStage?.minutes?.minutes
+        broodmotherSpawnTime = SimpleTimeMark.now() + timeUntilSpawn!!
+
+        if (currentStage == StageEntry.IMMINENT && config.imminentWarning) {
+            playImminentWarning()
+            return
+        }
+
+        if (config.stages.contains(currentStage)) {
+            if (currentStage == StageEntry.SLAIN) {
+                onBroodmotherSlain()
+            } else {
+                val pluralize = StringUtils.pluralize(timeUntilSpawn.toInt(DurationUnit.MINUTES), "minute")
+                ChatUtils.chat(
+                    "Broodmother: $lastStage §e-> $currentStage§e. §b${timeUntilSpawn.inWholeMinutes} $pluralize §euntil it spawns!"
+                )
+            }
+        }
+    }
+
+    private fun onServerJoin() {
+        if (config.stageOnJoin && currentStage != StageEntry.ALIVE) {
```

Comment:
Why dont we want to send this message when alive?

Reply:
So that the clickToActionOrDisable message is sent instead. Otherwise, if you joined a server where the Broodmother is alive you'd see this in chat:

```
[SkyHanni] The Broodmother's current stage in this server is Alive!
[SkyHanni] The Broodmother has spawned!
[CLICK to warp to the Spider Mound or disable this feature]"
```

Reply:
Oh, yeah. thanks for explanation. maybe wanna explain that in the line above in a comment? just break it down more